test_common/harness/kernelHelpers.c

```diff
@@ -74,17 +75,10 @@ std::vector<char> get_file_content(const std::string &fileName)
 
 std::string get_kernel_name(const std::string &source)
 {
-    cl_uint crc = 0;
+    cl_uint crc = 0x04c11db7;
```

Comment:
The value of `crc` here of `0x04c11db7` acts as a seed to the CRC32. IMO, it would be better not to have the seed, as the CRC would then be the same as produced by e.g. `jacksum -x -a fcs32`.

To remove the seed, the `crc` argument to `crc32()` should be removed, and the `crc` variable within `crc32` should be initialized with the value of `~0U`, exactly as per the example originally given in `libkern/crc32.c`:

* https://github.com/freebsd/freebsd/blob/master/sys/libkern/crc32.c#L122

I note that `0x04c11db7` is the normal representation of the polynomial for this CRC. However, the polynomial is applied through the use of `crc32_tab`, which has been generated from the polynomial. Altering the initial state of the function does not influence the polynomial that is used, but is instead equivalent to adding a prefix to the data for which the CRC is to be generated, which makes the value of `0x04c11db7` somewhat confusing. Any other non-zero value would suffice equally, although I would recommend, as stated above, not to use a seed value at all.

Reply:
This is not my code personally, but I guess that this seed was a kind of protection against nulls numbers in input data. Do not you think that having this seed the results of crc calculations will be more 'randomized'?

Reply:
> This is not my code personally, but I guess that this seed was a kind of protection against nulls numbers in input data. Do not you think that having this seed the results of crc calculations will be more 'randomized'?

It is true that the FCS-32 of the empty string is zero. However, I don't really see this as a problem. If anything, if the CTS were ever to create such a file, the hash of "00000000" would be more likely to stand out when looking at log output or a directory listing.

The "seed" is equivalent to prepending the data with a fixed, arbitrary sequence of bytes, and so offers no extra protection of any form. In this case, it happens to be equivalent to prepending the data with a byte sequence of: 0xd5, 0xe3, 0x1c, 0x09.

It seems unlikely that this behaviour was intentional.

Reply:
I see that this change force us regenerate spirv files we are using now. But if there is no other way to have this PR accepted we will do this :)

Reply:
Perhaps you can rename the SPIR-V files instead of regenerating them?

For example:
```
#!/bin/bash

find -name '*.cl' | while read FILE
do
    CHECKSUM=$(jacksum -x -a fcs32 -F '#CHECKSUM' "${FILE}")
    NEW_BASE_NAME=$(sed -r -e 's/^(.*)\.[^.]+\.([^.])+\.cl$/\1.'"${CHECKSUM}"'.\2/' <<< "${FILE}")
    for SUFFIX in bin bc bc32 bc64 spv spv32 spv64 options
    do
        if [[ -e "${FILE%.cl}.${SUFFIX}" ]]
        then
            mv -v "${FILE%.cl}.${SUFFIX}" "${NEW_BASE_NAME}.${SUFFIX}"
        fi
    done
    mv -v "${FILE}" "${NEW_BASE_NAME}.cl"
done
```

Reply:
Nice :) thank you very much